### ISSUE TITLE
error: Could not resolve "fs"

### ISSUE THREAD

**Issue by frading** [evanw/esbuild#90]
My library includes packages (`opentype` that is in threejs, and `ammo.js`) which use `require`, `path` and `fs`.

When compiling with webpack, I add the following to the config:
```
node: {
	fs: 'empty'
},
```

But how would you solve this with esbuild? Here is below the errors that are printed out:

```
modules/three/examples/jsm/libs/opentype.module.min.js:19:457: error: "require" must not be called indirectly
...ports;return (function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.co...
                                                                                                         ~~~~~~~
modules/three/examples/jsm/libs/opentype.module.min.js:19:757: error: "require" must not be called indirectly
...,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
                                                                                                                             ~~~~~~~
node_modules/ammojs-typed/ammo/ammo.js:11:83: error: Could not resolve "fs"
if(ea){ia=__dirname+"/";var na,oa;ka=function(a,b){var e=pa(a);e||(na||(na=require("fs")),oa||(oa=require("path")),a=oa.normalize(a),e=na.readFileSync(a));return b?e:e.toString()};ma=function(a){a=ka(a,!0);a.buffe...
                                                                                   ~~~~
node_modules/ammojs-typed/ammo/ammo.js:11:106: error: Could not resolve "path"
...a){ia=__dirname+"/";var na,oa;ka=function(a,b){var e=pa(a);e||(na||(na=require("fs")),oa||(oa=require("path")),a=oa.normalize(a),e=na.readFileSync(a));return b?e:e.toString()};ma=function(a){a=ka(a,!0);a.buffer...
                                                                                                         ~~~~~~
4 errors
```

And here is how to reproduce:

``` bash
git clone https://github.com/polygonjs/polygonjs-engine.git;
cd polygonjs-engine;
git checkout esbuild;
yarn;
node_modules/esbuild/bin/esbuild src/engine/index.ts --bundle '--define:process.env.NODE_ENV="development"' --loader:.glsl=text --outfile=out.js;
```

**Comment by jorgelo** [comment]
I found that when using it via the API, the build method can take in parameter of `externals`. This seems to exclude them from the build.

**Comment by frading** [comment]
@jorgelo thank you, that does get rid of the `fs` and `path` errors. But I haven't found a way to get rid of opentype's `require` yet.

Here is the current build script I'm using:
```
import {build} from 'esbuild';
import {BuildOptions} from 'esbuild/lib/main';

const out = './dist/out.js';

const options: BuildOptions = {
	stdio: 'inherit',
	entryPoints: ['./src/engine/index.ts'],
	target: 'esnext',
	outfile: out,
	minify: true,
	bundle: true,
	define: {
		'process.env.NODE_ENV': '"development"',
	},
	external: ['require', 'fs', 'path'],
	loader: {
		'.glsl': 'text',
	},
};

build(options).catch(() => process.exit(1));
```

**Comment by evanw** [comment]
This should be fixed in version 0.5.3. It turns out the problem was specifically caused by code bundled with Browserify, so I have special-cased this code pattern in esbuild. You can read more in the [release notes](https://github.com/evanw/esbuild/blob/master/CHANGELOG.md#053).

**Comment by frading** [comment]
Amazing, thank you!
